procedures/MT/m2/m2_bump_test.ipynb

```diff
@@ -0,0 +1,217 @@
+    "    await csc.cmd_applyForces.set_start(axial=forces)\n",
+    "    await asyncio.sleep(sleep_time)\n",
+    "\n",
+    "    # Put back to origin\n",
+    "    print(\"Reset the force.\")\n",
+    "    await csc.cmd_resetForceOffsets.set_start()\n",
+    "    await asyncio.sleep(sleep_time)\n",
+    "\n",
+    "    # Do the Negative direction\n",
+    "    forces[idx_actuator] = -abs(force)\n",
+    "\n",
+    "    print(f\"Apply the force: {-abs(force)} N.\")\n",
+    "    await csc.cmd_applyForces.set_start(axial=forces)\n",
+    "    await asyncio.sleep(sleep_time)\n",
+    "\n",
+    "    # Put back to origin\n",
+    "    print(\"Reset the force.\")\n",
+    "    await csc.cmd_resetForceOffsets.set_start()\n",
+    "    await asyncio.sleep(sleep_time)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "79c53e71-d4dc-4e40-b856-b0af2a045b4d",
+   "metadata": {},
+   "source": [
+    "## Prepare the M2 CSC and put to Enabled state"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "146c08d7-48a0-4d41-b512-419bb6bb87d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "domain = salobj.Domain()\n",
+    "m2 = salobj.Remote(domain, \"MTM2\")\n",
+    "await m2.start_task\n",
+    "await m2.cmd_setLogLevel.set_start(level=10)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "fffd6ad9-6f0f-40c3-b00a-3655bf5a6a1c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "state_m2 = m2.evt_summaryState.get()\n",
+    "if state_m2.summaryState != salobj.State.ENABLED:\n",
+    "    await salobj.set_summary_state(m2, salobj.State.ENABLED, timeout=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5961eb58-cbbb-48b6-bdc7-e6518faebb3a",
+   "metadata": {},
+   "source": [
+    "## Do the Bump Test"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "87537214-5f5f-4f41-aadf-0cb26df49b1f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "time_start = datetime.now()\n",
+    "print(f\"UTC time to is {time_start} now.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8a9380fb-5c00-422e-9aae-32793598caab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "force = 10\n",
+    "await run_bump_test(m2, force, sleep_time=5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "276a342e-24ec-428b-9bd1-cc0d9d83b6e7",
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "LSST",
+   "language": "python",
+   "name": "lsst"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.10.6"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
I Hav two questions for you Te-Wei regarding this Script:

1)  In lines 64 to 68  you define a "zenithAngle" which seems to be the only elevation angle you use for this Test. Then, Do this script only works to perform a Bump Test for the Zenith Angle and not for other Elevation Angles?

2)  It is not clear to me if when you apply a Force to one of the actuators at the time you also check what happens to the other 6x or fewer "surrounding actuators" as well?

Reply:
1. For the "zenithAngle" telemetry, it is defined in the xml that contains multiple fields:
https://github.com/lsst-ts/ts_xml/blob/develop/python/lsst/ts/xml/data/sal_interfaces/MTM2/MTM2_Telemetry.xml#L204-L229

It just prints the angle as the information in data analysis. The bump test can be performed at any zenith angle theoretically.

2. I do not understand your question. The bump test is to apply the force to a single actuator at a time. For the neighboring actuators, we are interested in their behaviors and we will analyze the data after the bump test is done.